spec/facility_spec.rb

```diff
@@ -24,3 +24,237 @@
+    @facility_2 = Facility.new({name: 'Ashland DMV Office', address: '600 Tolman Creek Rd Ashland OR 97520', phone: '541-776-6092' })
+  end
+  describe "Iteration 2" do
+    it 'Registrants and Facilities exist' do
+      expect(@registrant_1).to be_an_instance_of(Registrant) 
+      expect(@registrant_2).to be_an_instance_of(Registrant) 
+      expect(@registrant_3).to be_an_instance_of(Registrant)
+      expect(@facility_1).to be_an_instance_of(Facility)
+      expect(@facility_2).to be_an_instance_of(Facility)
+    end
+
+    # ---------- WRITTEN TEST ----------
+
+    it 'cannot administer written test without service' do
+      expect(@registrant_1.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+      expect(@registrant_1.permit?).to eq true
+      @facility_1.administer_written_test(@registrant_1)
+      expect(@facility_1.services).to eq([])
+      expect(@registrant_1.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+    end
+
+    it 'now has Written Test Service and can administer test' do
+      @facility_1.add_service('Written Test')
+      expect(@facility_1.services).to eq(['Written Test'])
+
+      @facility_1.administer_written_test(@registrant_1)
+      expect(@registrant_1.license_data).to eq({:written=>true, :license=>false, :renewed=>false})
+    end
+
+    it 'can administer the Written Test to another registrant' do
+      @facility_1.add_service('Written Test')
+      expect(@registrant_2.age).to eq(16)
+      expect(@registrant_2.permit?).to eq false
+
+      @facility_1.administer_written_test(@registrant_2)
+      expect(@registrant_2.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+      
+      @registrant_2.earn_permit
+      
+      @facility_1.administer_written_test(@registrant_2)
+      expect(@registrant_2.license_data).to eq({:written=>true, :license=>false, :renewed=>false})
+    end
+
+    it 'cannot administer Written Test to someone under 15 even with permit' do
+      @facility_1.add_service('Written Test')
+      expect(@registrant_3.age).to eq(15)
+      expect(@registrant_3.permit?).to eq false
+
+      @facility_1.administer_written_test(@registrant_3)
+      expect(@registrant_3.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+      
+      @registrant_3.earn_permit
+      
+      @facility_1.administer_written_test(@registrant_3)
+      expect(@registrant_3.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+    end
+    
+    # ---------- ROAD TEST ----------
+    
+    it 'cannot administer Road Test without service
+    or to registrant without permit and written test' do
+    @facility_1.add_service('Written Test')
+    expect(@facility_1.services).to eq(['Written Test'])
+    @facility_1.add_service('Road Test')
+    @facility_1.administer_road_test(@registrant_3)
+    expect(@facility_1.administer_road_test(@registrant_3)).to eq false
+    
```

Comment:
Good test here for testing that the method does nothing before the service has been added.